### PR TITLE
Add mantidpython launch script to MantidPlot application directory on OSX

### LINKED ISSUE #13728: Create launchers for ipython and ipython notebooks in OS X
We now have shortcuts in Windows which launch ipython or ipython notebook using the Mantid python environment.
Implement this idea for OS X. This may be possible by creating an application which contains a single bash script to import the mantid python environment and launch ipython.

### PR DESCRIPTION
**Pull request by matthew-d-jones**
Closes #13728 

Added a `mantidpython` script to the `MantidPlot.app/Contents/MacOS/`. This is equivalent to the script of the same name in the `bin` directory of linux installations of Mantid. It can be used to launch python using Mantid's python environment.

I have also added launcher applications for ipython console and ipython notebooks. These can optionally be installed from the MantidPlot dmg image. As they are applications they can be added to the dock etc.

**To Test:**

Must be tested on OS X.

I've put the dmg file from the build server at `smb://olympic/babylon5/Scratch/Matthew Jones` (commit ed1ac08).

Please check that the launchers work. The `mantidpython` script is used by the launchers, so testing the launchers also tests the script.

### CODE REVIEW

**Comment by matthew-d-jones** [comment]
I've updated http://download.mantidproject.org/osx.html

File: buildconfig/CMake/Packaging/osx/mantidpython_osx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+#
+# For OS X only
+# Launch IPython with paths set for the Mantid environment
+
+# Find out where we are
+INSTALLDIR=/Applications/MantidPlot.app
```
Comment (quantumsteve) [inline buildconfig/CMake/Packaging/osx/mantidpython_osx]: If possible, we should avoid using absolute paths in the app bundle. A user should be able to install and run Mantid in another directory, such as `~/Applications`.

**Comment by matthew-d-jones** [comment]
@quantumsteve Thanks for your comment on the absolute paths. This was easily solved for the `mantidpython` script, but not for the launcher applications.
The best I've come up with allows them to work wherever they are installed, as long as they are installed in the same place as MantidPlot. Do you think this is acceptable?

**Comment by quantumsteve** [comment]
@matthew-d-jones Yes, I think that is acceptable. The user can create a symlink to the bundled file in a more convenient place.

**Comment by matthew-d-jones** [comment]
@quantumsteve  Ok, thanks.
@OwenArnold In that case the PR is ready for testing. I've put the dmg on the network drive in case it is not available on the build server when you come to test it: `smb://olympic/babylon5/Scratch/Matthew Jones`

**Comment by OwenArnold** [comment]
All seems to work. Tested putting the installer a user directory, and all worked fine there too.

**Pull request merged**